018_go_list_mod_graph_why/script.sh

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env bash
+
+# **START**
+
+export GOPATH=$HOME
+export PATH=$GOPATH/bin:$PATH
+echo "machine github.com login $GITHUB_USERNAME password $GITHUB_PAT" >> $HOME/.netrc
+echo "" >> $HOME/.netrc
+echo "machine api.github.com login $GITHUB_USERNAME password $GITHUB_PAT" >> $HOME/.netrc
+git config --global user.email "$GITHUB_USERNAME@example.com"
+git config --global user.name "$GITHUB_USERNAME"
+git config --global advice.detachedHead false
+git config --global push.default current
+
+mkdir $HOME/scratchpad
+
+now=$(date +'%Y%m%d%H%M%S_%N')
+githubcli repo create $GITHUB_ORG/listmodgraphwhy_analysis_new_$now
+
+# build listmodwhygraph and incomplete
+githubcli repo renameIfExists $GITHUB_ORG/listmodwhygraph listmodwhygraph_$now
+githubcli repo transfer $GITHUB_ORG/listmodwhygraph_$now $GITHUB_ORG_ARCHIVE
+githubcli repo create $GITHUB_ORG/listmodwhygraph
+githubcli repo renameIfExists $GITHUB_ORG/incomplete incomplete_$now
+githubcli repo transfer $GITHUB_ORG/incomplete_$now $GITHUB_ORG_ARCHIVE
+githubcli repo create $GITHUB_ORG/incomplete
+
+comment
+comment "Prepare incomplete"
+comment
+
+cd $(mktemp -d)
+git init
+git remote add origin https://github.com/$GITHUB_ORG/incomplete
+cat <<EOD > go.mod
+module github.com/$GITHUB_ORG/incomplete
+EOD
+go mod edit -fmt
+cat <<EOD > incomplete.go
+package incomplete
+
+import (
+	_ "github.com/sirupsen/logrus"
+	_ "golang.org/x/tools/go/packages"
+)
+EOD
+gofmt -w incomplete.go
+git add -A
+git commit -am 'Initial commit'
+git push
+
+comment
+comment "Prepare listmodwhygraph"
+comment
+
+cd $(mktemp -d)
+git init
+git remote add origin https://github.com/$GITHUB_ORG/listmodwhygraph
+cat <<EOD > go.mod
+module github.com/$GITHUB_ORG/listmodwhygraph
+EOD
+go mod edit -fmt
+cat <<EOD > listmodwhygraph.go
+package listmodwhygraph
+
+import (
+	_ "github.com/$GITHUB_ORG/incomplete"
+	_ "github.com/kr/pretty"
+	_ "github.com/sirupsen/logrus"
+)
+EOD
+gofmt -w listmodwhygraph.go
+go mod tidy
+git add -A
+git commit -am 'Initial commit'
+git push
+
+comment
+comment "====================="
+comment
+
+export baserepo=https://github.com/$GITHUB_ORG/listmodwhygraph
+
+# egrunner_envsubst: +mainmod +baserepo
+
+# block: setup
+git clone -q $baserepo $HOME/scratchpad/listmodwhygraph
+cd $HOME/scratchpad/listmodwhygraph
+
+# block: base repo
+echo $baserepo
+
+# block: org
+echo $GITHUB_ORG
+
+# block: repo
+echo https://github.com/$GITHUB_ORG/listmodgraphwhy_analysis_new_$now
+
+# behind the scenes download
+go mod download
+
+export mainmod=$(go list -m)
+
+# block: main module
+go list -m
+
+# block: main module pkgs
+go list $mainmod/...
+
+# block: pkgpath
+go list
+
+# block: tree main
+tree --noreport
+
+# block: listmodwhygraph.go
+catfile $PWD/listmodwhygraph.go
```

Comment:
equally, I'm not sure why some of these commands use absolute paths, when others rely on the implicit $PWD :)

Reply:
`catfile` is a special function. I'm using `$PWD` here in order to remind the user "where" we are. With so many commands, some of which change directory, it can be easy to forget where you are. So I switched to this approach in order to remove that ambiguity. And it's also why, incidentally, things are generally done within `$HOME/scratchpad`; it's the very clear to the user that we're doing something "throwaway". `/tmp` would be somewhat confusing because if people are following along with the guide, they'll likely want to do so in an area that they don't then lose on the next restart :)